File: src/main/java/com/example/checkoutplanner/entity/Wishbook.java
package com.example.checkoutplanner.entity;

import jakarta.persistence.*;
import lombok.Data;

@Entity
@Data
public class Wishbook {

    public enum Shift {
        SHIFT_1,
        SHIFT_2
    }

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String date;
    private String employeeId;

    @Enumerated(EnumType.STRING)
    private Shift shift;
}

File: frontend/lib/models/wishbook.dart
class Wishbook {
  final int id;
  final String date;
  final String employeeId;
  final String shift;

  Wishbook({
    required this.id,
    required this.date,
    required this.employeeId,
    required this.shift
  });

  factory Wishbook.fromJson(Map<String, dynamic> json) {
    return Wishbook(
        id: json["id"],
        date: json["date"],
        employeeId: json["employeeId"],
        shift: json["shift"]
    );
  }

  Map<String, dynamic> toJson() {
    return {
      "id": id,
      "date": date,
      "employeeId": employeeId,
      "shift": shift
    };
  }
}